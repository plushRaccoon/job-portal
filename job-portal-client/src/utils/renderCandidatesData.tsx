// import React, { useState } from 'react';
// import { List, ListItem, IconButton, ListItemText, Box, ListItemIcon, TextField, Button, Modal } from "@mui/material";
// import Divider from "@mui/material/Divider";
// import DeleteIcon from '@mui/icons-material/Delete';
// import BorderColorRoundedIcon from '@mui/icons-material/BorderColorRounded';
// import { StyleForModals } from './constants';
// import EditApplicationForm from '../components/application-form';

// export const RenderCandidatesData = ({ item }) => (
//   <>
//     {item.firstName} {item.lastName}
//     <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
//     {item.email}
//     <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
//     {item.phone}
//   </>
// )

// export const RenderCandidatesDetails = ({ item, onDelete, onEdit }) => {
//   // const [editedCV, setEditedCV] = useState('');
//   // const [openModal, setOpenModal] = useState(false);
//   // const [ applicationId, setApplicationId ] = useState('');
//   const [openEditModal, setOpenEditModal] = useState(false);
//   const [editApplication, setEditApplication] = useState(null);

//   // const handleEditClick = (id, cv) => {
//   //   setOpenModal(true);
//   //   setApplicationId(id);
//   //   setEditedCV(cv);
//   // };

//   // const handleCancelEdit = () => {
//   //   setOpenModal(false);
//   //   setEditedCV('');
//   // };

//   // const handleSubmitEdit = (editedCV) => {
//   //   onEdit('applications', applicationId, editedCV);
//   //   setOpenModal(false);
//   //   setEditedCV('');
//   // };

//   const handleEditClick = (application) => {
//     setEditApplication(application);
//     setOpenEditModal(true);
//   };

//   const handleCloseEditModal = () => {
//     setOpenEditModal(false);
//     setEditApplication(null);
//   };

//   const handleSubmitEdit = (id, editedData) => {
//     onEdit('applications', id, editedData);
//     setOpenEditModal(false);
//     setEditApplication(null);
//   };

//   return (
//     <>
//       {item.applications?.map((application) => {
//         const { id, cv, position: { title, status } } = application;
//         return (
//           <List key={id}>
//             <ListItem secondaryAction={
//               <IconButton edge="end" aria-label="delete" color='error' onClick={() => onDelete('applications', id)}>
//                 <DeleteIcon />
//               </IconButton>
//             }>
//               <ListItemText sx={{ marginRight: 10 }}>
//                 <Box key={id}>
//                   <Box display="flex" alignItems="center" marginBottom="10px">
//                     {title}
//                     <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
//                     {status}
//                   </Box>
//                   {cv}
//                 </Box>
//               </ListItemText>
//               <ListItemIcon>
//                   {/* <IconButton size="large" aria-label="edit" color="default" onClick={() => handleEditClick(id, cv)}> */}
//                   <IconButton size="large" aria-label="edit" color="default" onClick={() => handleEditClick(application)}>
//                     <BorderColorRoundedIcon />
//                   </IconButton>
//               </ListItemIcon>
//             </ListItem>
//           </List>
//         );
//       })}
//       {editApplication && (
//         <EditApplicationForm
//           open={openEditModal}
//           handleClose={handleCloseEditModal}
//           application={editApplication}
//           handleSubmit={handleSubmitEdit}
//         />
//       )}
//       {/* <Modal
//         open={openModal}
//         onClose={handleCancelEdit}
//         aria-labelledby="edit-cv-form-title"
//         aria-describedby="edit-cv-form-description"
//       >
//         <Box sx={StyleForModals}>
//           <TextField
//             margin="normal"
//             required
//             fullWidth
//             multiline
//             id="cv"
//             label="CV"
//             name="cv"
//             value={editedCV}
//             onChange={(e) => setEditedCV(e.target.value)}
//           />
//           <Button onClick={() => handleSubmitEdit(editedCV)} variant="contained" color="primary" fullWidth size='large'>
//             Submit
//           </Button>
//         </Box>
//       </Modal> */}
//     </>
//   );
// };


import { useState } from 'react';
import { Box, List, ListItem, ListItemText, ListItemIcon, IconButton, Divider } from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete';
import BorderColorRoundedIcon from '@mui/icons-material/BorderColorRounded';
import EditApplicationForm from '../components/application-form'; // Ensure the import is correct

export const RenderCandidatesData = ({ item }) => (
    <>
      {item.firstName} {item.lastName}
      <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
      {item.email}
      <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
      {item.phone}
    </>
  )

export const RenderCandidatesDetails = ({ item, onDelete, onEdit }) => {
  const [openEditModal, setOpenEditModal] = useState(false);
  const [editApplication, setEditApplication] = useState(null);

  const handleEditClick = (application) => {
    setEditApplication(application);
    setOpenEditModal(true);
  };

  const handleCloseEditModal = () => {
    setOpenEditModal(false);
    setEditApplication(null);
  };

  const handleSubmitEdit = (id) => {
    onEdit('applications', id, editApplication);
    setOpenEditModal(false);
    setEditApplication(null);
  };

  return (
    <>
      {item.applications?.map((application) => {
        const { id, cv, position: { title, status } } = application;
        return (
          <List key={id}>
            <ListItem
              secondaryAction={
                <IconButton edge="end" aria-label="delete" color="error" onClick={() => onDelete('applications', id)}>
                  <DeleteIcon />
                </IconButton>
              }
            >
              <ListItemText sx={{ marginRight: 10 }}>
                <Box key={id}>
                  <Box display="flex" alignItems="center" marginBottom="10px">
                    {title}
                    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
                    {status}
                  </Box>
                  {cv}
                </Box>
              </ListItemText>
              <ListItemIcon>
                <IconButton size="large" aria-label="edit" color="default" onClick={() => handleEditClick(application)}>
                  <BorderColorRoundedIcon />
                </IconButton>
              </ListItemIcon>
            </ListItem>
          </List>
        );
      })}
      {editApplication && (
        <EditApplicationForm
          open={openEditModal}
          handleClose={handleCloseEditModal}
          application={editApplication}
          handleSubmit={() => handleSubmitEdit(editApplication.id)}
        />
      )}
    </>
  );
};
