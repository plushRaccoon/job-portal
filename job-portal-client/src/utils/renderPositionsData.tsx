import React from "react";
import { List, ListItem, ListItemText, Box } from "@mui/material";
import Divider from "@mui/material/Divider";

export const RenderPositionsData = ({ item }) => {
  return (
  <>
    {item.title}
    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
    {item.status}
    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
    Applied: {item.applications?.length || 0}
  </>
)}

export const RenderPositionsDetails = ({item}) => {
  return (
  <>
    {item.applications?.map((application) => {
      const { candidate: { firstName, lastName, email, phone }, cv, id } = application;
      return (
        <List key={id}>
          <ListItem>
            <ListItemText>
                <Box key={id}>
                  <Box display="flex" alignItems="center" marginBottom="10px">
                    {firstName} {lastName}
                    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
                    {email}
                    <Divider orientation="vertical" flexItem style={{ margin: '0 8px' }} />
                    {phone}
                  </Box>
                  {cv}
                </Box>
            </ListItemText>
          </ListItem>
        </List>
      );
    })}
  </>
)}
