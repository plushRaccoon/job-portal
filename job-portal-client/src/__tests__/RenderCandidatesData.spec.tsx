import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { RenderCandidatesData, RenderCandidatesDetails  } from '../utils/renderCandidatesData';
import '@testing-library/jest-dom';
import { vi } from 'vitest';


const mockApplication = {
  id: '1',
  cv: 'John Doe CV',
  position: {
    title: 'Software Engineer',
    status: 'Active',
  },
};

const mockItem = {
  applications: [mockApplication],
};

describe('RenderCandidatesData Component', () => {
  const mockItem = {
    firstName: 'John',
    lastName: 'Doe',
    email: 'john.doe@example.com',
    phone: '123-456-7890',
  };

  it('matches snapshot', () => {
    const { asFragment } = render(<RenderCandidatesData item={mockItem} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('RenderCandidatesDetails Component', () => {
  const mockOnDelete = vi.fn();
  const mockOnEdit = vi.fn();

  it('renders applications correctly', () => {
    render(<RenderCandidatesDetails item={mockItem} onDelete={mockOnDelete} onEdit={mockOnEdit} />);

    const component = screen.getByRole('listitem');
    expect(component).toMatchSnapshot();
  });

  it('handles delete button click', () => {
    render(<RenderCandidatesDetails item={mockItem} onDelete={mockOnDelete} onEdit={mockOnEdit} />);

    fireEvent.click(screen.getByLabelText('delete'));
    expect(mockOnDelete).toHaveBeenCalledWith('applications', mockApplication.id);
  });

  it('opens edit modal on edit button click', () => {
    render(<RenderCandidatesDetails item={mockItem} onDelete={mockOnDelete} onEdit={mockOnEdit} />);

    fireEvent.click(screen.getByTestId('BorderColorRoundedIcon'));
    expect(screen.getByText('Edit Application')).toBeInTheDocument();
  });

  it('handles edit form submission', () => {
    render(<RenderCandidatesDetails item={mockItem} onDelete={mockOnDelete} onEdit={mockOnEdit} />);

    fireEvent.click(screen.getByLabelText('edit'));
    fireEvent.click(screen.getByText('Save Changes'));

    expect(mockOnEdit).toHaveBeenCalledWith('applications', mockApplication.id, mockApplication);
  });
});
