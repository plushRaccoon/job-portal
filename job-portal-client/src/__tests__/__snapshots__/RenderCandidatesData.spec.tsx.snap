// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RenderCandidatesData Component > matches snapshot 1`] = `
<DocumentFragment>
  John Doe
  <hr
    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
    style="margin: 0px 8px;"
  />
  john.doe@example.com
  <hr
    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
    style="margin: 0px 8px;"
  />
  123-456-7890
</DocumentFragment>
`;

exports[`RenderCandidatesDetails Component > renders applications correctly 1`] = `
<li
  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
>
  <div
    class="MuiListItemText-root css-1vculsn-MuiListItemText-root"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1rwzck"
        >
          Software Engineer
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
            style="margin: 0px 8px;"
          />
          Active
        </div>
        John Doe CV
      </div>
    </span>
  </div>
  <div
    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
  >
    <button
      aria-label="edit"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="BorderColorRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 24H4c-1.1 0-2-.9-2-2s.9-2 2-2h16c1.1 0 2 .9 2 2s-.9 2-2 2M13.06 5.19l3.75 3.75-8.77 8.77c-.18.19-.44.29-.7.29H5c-.55 0-1-.45-1-1v-2.34c0-.27.11-.52.29-.71zm4.82 2.68-3.75-3.75 1.83-1.83c.39-.39 1.02-.39 1.41 0l2.34 2.34c.39.39.39 1.02 0 1.41z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
  >
    <button
      aria-label="delete"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-gmng1k-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</li>
`;
